### PR TITLE
Team stats

### PR DESCRIPTION
**Pull request by yongning0310**
Display team and player stats during battle. 

A button that acts as a correct submission, updating team score, player score, and questions solved. 

A button that acts as a wrong submission, deducting player health. 

When a round ends, reset all stats, including health and round scores, and update the winner's match score

### CODE REVIEW

File: client/src/scenes/Battle.ts [hunk 1]
```diff
@@ -116,39 +118,98 @@ export default class Battle extends Phaser.Scene {
   }
 
 
-  // set up the team listener to display the team  when teams.onChange 
+  private setUpQuestions() {
+    let correctAnswerButton = this.add.text(0, 150, 'Correct Answer', {
```
Comment (shashahchk) [inline client/src/scenes/Battle.ts]: QuestionLogic file?

Comment (shashahchk) [reply client/src/scenes/Battle.ts]: correctAnswer() can be abstracted awya in that file
separate out ui components

File: client/src/scenes/Battle.ts [hunk 2]
```diff
@@ -116,39 +118,98 @@ export default class Battle extends Phaser.Scene {
...
+    let wrongAnswerButton = this.add.text(0, 190, 'Wrong Answer', {
+      fontSize: '20px',
+      backgroundColor: '#f00',
+      fixedWidth: 200
+    })
+      .setInteractive()
+      .on('pointerdown', () => this.wrongAnswer());
+  }
+
+  // this will always be correct
```
Comment (shashahchk) [inline client/src/scenes/Battle.ts]: sendCheckAnswer in the real implementation

File: client/src/scenes/Battle.ts [hunk 3]
```diff
@@ -116,39 +118,98 @@ export default class Battle extends Phaser.Scene {
...
   private setUpTeamListeners() {
-    // on message for "teamUpdate"
     this.room.onMessage("teamUpdate", (message) => {
       const teamList = message.teams;
       console.log("Team update", teamList);
 
-      let teamTexts = teamList.map((team, index) => {
+      let allInfo = ""
+      let specificPlayer = null;
+      let specificPlayerInfo = "";
```
Comment (shashahchk) [inline client/src/scenes/Battle.ts]: current player?

File: client/src/scenes/Battle.ts [hunk 4]
```diff
@@ -116,39 +118,98 @@ export default class Battle extends Phaser.Scene {
...
         if (team && typeof team === 'object') {
           let teamColor = team.teamColor;
           let teamPlayersNames = [];
 
-          // Assuming 'teamPlayers' is an object where keys are player IDs and values are player objects
           for (let playerId in team.teamPlayers) {
             if (team.teamPlayers.hasOwnProperty(playerId)) {
               let player = team.teamPlayers[playerId];
-              teamPlayersNames.push(playerId); // need to change to "name" property of player
+              teamPlayersNames.push(playerId);
```
Comment (shashahchk) [inline client/src/scenes/Battle.ts]: player.userName instead, be careful of the logic when naming

File: server/src/rooms/BattleRoom.ts [hunk 5]
```diff
@@ -83,6 +129,41 @@ export class BattleRoom extends Room<BattleRoomState> {
...
+    } else {
+      // If there are no teams with a max score, increment the match score for all teams
+      this.state.teams.forEach(team => {
+        team.teamMatchScore += 1;
+      });
+    }
+
+    this.state.teams.forEach((team) => {
+      team.teamRoundScore = 0;
+      team.teamPlayers.forEach((player) => {
```
Comment (shashahchk) [inline server/src/rooms/BattleRoom.ts:169]: abstract out and name method

File: server/src/rooms/BattleRoom.ts [hunk 6]
```diff
@@ -83,6 +129,41 @@ export class BattleRoom extends Room<BattleRoomState> {
...
+        player.roundQuestionIdsSolved = new ArraySchema<number>();
+        player.roundScore = 0;
+        player.health = 100;
```
Comment (shashahchk) [inline server/src/rooms/BattleRoom.ts]: have constant for maxPlayerHealth

File: server/src/rooms/schema/Character.ts [hunk 7]
```diff
@@ -31,9 +31,18 @@ export class Monster extends Character {
...
   @type("number") roundScore: number;
+  @type(["number"]) roundQuestionIdsSolved: ArraySchema<number>;
   @type("string") teamColor: TeamColor;
-  @type("boolean") isAlive: boolean;
-  @type(["number"]) questionIdsSolved: number; // abit sus but we'll see
   @type(Monster) monster: Monster;
+
+  constructor() {
+    super();
+    this.health = 100;
```
Comment (shashahchk) [inline server/src/rooms/schema/Character.ts]: dont hardcode pls, very hard to find and change later on

**Review by pygmychameleon — CHANGES_REQUESTED** [review]
LGTM. However, logic will have to be changed majorly when the questions are integrated since the checkCorrect or Wrong should not be handled on client side. They should not choose what method to call. 
Also, try to have more abstractions.

**Pull request merged**